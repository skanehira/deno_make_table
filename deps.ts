export { Table } from "https://deno.land/x/cliffy@v0.25.7/table/mod.ts";
export { assertSnapshot } from "https://deno.land/std@0.175.0/testing/snapshot.ts";
