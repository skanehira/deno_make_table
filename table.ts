import { Table } from "./deps.ts";

export function makeTable(input: string): string {
  const lines = input.split("\n").filter((v) => v.trim());
  const header = lines.shift()!.split(",").map((v) => v.trim());
  const body = lines.map((line) => line.split(",").map((l) => l.trim()));

  return new Table()
    .header(header)
    .body(body)
    .maxColWidth(15)
    .padding(1)
    .border(true)
    .toString();
}
