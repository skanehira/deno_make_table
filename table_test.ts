import { makeTable } from "./table.ts";
import { assertSnapshot } from "./deps.ts";

Deno.test({
  name: "make table",
  fn: async (t) => {
    const input = `h1, h2
col1, col2
col3, col4
    `;

    const got = makeTable(input);
    await assertSnapshot(t, got);
  },
});
