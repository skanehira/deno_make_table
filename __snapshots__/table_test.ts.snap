export const snapshot = {};

snapshot[`make table 1`] = `
"┌──────┬──────┐
│ h1   │ h2   │
├──────┼──────┤
│ col1 │ col2 │
├──────┼──────┤
│ col3 │ col4 │
└──────┴──────┘"
`;
